### PR TITLE
Bigint update case

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
It appeared that old-fashioned MDB2 schema ignores changing length and doesn't migrate it to bigint.
Due to issue with migrations on postgres minor supported version for apps was increased 

## Related Issue
https://github.com/owncloud/core/issues/26901

## How Has This Been Tested?
Manual update from 10.0.2

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by VicDeo** [comment]
WTF is that????
https://github.com/owncloud/core/blob/master/apps/files_sharing/appinfo/app.php#L64

**Comment by VicDeo** [comment]
A new bunch of surprises. `Files_sharing` is installed before `files` and contains another gift inside....

File: apps/federatedfilesharing/appinfo/Migrations/Version20170804201125.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		$prefix = $options['tablePrefix'];
+		if ($schema->hasTable("{$prefix}federated_reshares")) {
+			return;
+		}
+		// not that valid ....
+		$schemaReader = new MDB2SchemaReader(
+			\OC::$server->getConfig(),
+			\OC::$server->getDatabaseConnection()->getDatabasePlatform()
+		);
+		$schemaReader->loadSchemaFromFile(__DIR__ . '/../database.xml', $schema);
```
Comment (PVince81) [inline apps/federatedfilesharing/appinfo/Migrations/Version20170804201125.php]: just port the table to a migration ?

Comment (VicDeo) [inline apps/federatedfilesharing/appinfo/Migrations/Version20170804201125.php]: @PVince81 done

**Review by PVince81 — APPROVED** [review]
:+1: looks good

@VicDeo did you test installing, upgrading this time ?
Diffed the schema dumps to confirm that they are the same when using migrations ?

**Comment by VicDeo** [comment]
@PVince81 with Mysql only so far

**Comment by VicDeo** [comment]
```
mysql> use o1002;
Database changed
mysql> describe oc_federated_reshares;
+-----------+---------+------+-----+---------+-------+
| Field     | Type    | Null | Key | Default | Extra |
+-----------+---------+------+-----+---------+-------+
| share_id  | int(11) | NO   | PRI | NULL    |       |
| remote_id | int(11) | NO   |     | NULL    |       |
+-----------+---------+------+-----+---------+-------+
2 rows in set (0.00 sec)

mysql> use biginttest;
Database changed
mysql> describe oc_federated_reshares;
+-----------+------------+------+-----+---------+-------+
| Field     | Type       | Null | Key | Default | Extra |
+-----------+------------+------+-----+---------+-------+
| share_id  | bigint(20) | NO   | PRI | NULL    |       |
| remote_id | bigint(20) | NO   |     | NULL    |       |
+-----------+------------+------+-----+---------+-------+
2 rows in set (0.00 sec)

mysql> use o1002;
Database changed
mysql> describe oc_share_external;
+-----------------+---------------+------+-----+---------+----------------+
| Field           | Type          | Null | Key | Default | Extra          |
+-----------------+---------------+------+-----+---------+----------------+
| id              | int(11)       | NO   | PRI | NULL    | auto_increment |
| remote          | varchar(512)  | NO   |     | NULL    |                |
| remote_id       | int(11)       | NO   |     | -1      |                |
| share_token     | varchar(64)   | NO   |     | NULL    |                |
| password        | varchar(64)   | YES  |     | NULL    |                |
| name            | varchar(64)   | NO   |     | NULL    |                |
| owner           | varchar(64)   | NO   |     | NULL    |                |
| user            | varchar(64)   | NO   | MUL | NULL    |                |
| mountpoint      | varchar(4000) | NO   |     | NULL    |                |
| mountpoint_hash | varchar(32)   | NO   |     | NULL    |                |
| accepted        | int(11)       | NO   |     | 0       |                |
+-----------------+---------------+------+-----+---------+----------------+
11 rows in set (0.00 sec)
mysql> use biginttest;
Database changed
mysql> describe oc_share_external;
+-----------------+---------------+------+-----+---------+----------------+
| Field           | Type          | Null | Key | Default | Extra          |
+-----------------+---------------+------+-----+---------+----------------+
| id              | bigint(20)    | NO   | PRI | NULL    | auto_increment |
| remote          | varchar(512)  | NO   |     | NULL    |                |
| remote_id       | bigint(20)    | NO   |     | -1      |                |
| share_token     | varchar(64)   | NO   |     | NULL    |                |
| password        | varchar(64)   | YES  |     | NULL    |                |
| name            | varchar(64)   | NO   |     | NULL    |                |
| owner           | varchar(64)   | NO   |     | NULL    |                |
| user            | varchar(64)   | NO   | MUL | NULL    |                |
| mountpoint      | varchar(4000) | NO   |     | NULL    |                |
| mountpoint_hash | varchar(32)   | NO   |     | NULL    |                |
| accepted        | int(11)       | NO   |     | 0       |                |
+-----------------+---------------+------+-----+---------+----------------+
11 rows in set (0.00 sec)


mysql> use o1002;
Database changed
mysql> describe oc_files_trash;
+-----------+--------------+------+-----+---------+----------------+
| Field     | Type         | Null | Key | Default | Extra          |
+-----------+--------------+------+-----+---------+----------------+
| auto_id   | int(11)      | NO   | PRI | NULL    | auto_increment |
| id        | varchar(250) | NO   | MUL |         |                |
| user      | varchar(64)  | NO   | MUL |         |                |
| timestamp | varchar(12)  | NO   | MUL |         |                |
| location  | varchar(512) | NO   |     |         |                |
| type      | varchar(4)   | YES  |     | NULL    |                |
| mime      | varchar(255) | YES  |     | NULL    |                |
+-----------+--------------+------+-----+---------+----------------+
7 rows in set (0.00 sec)

mysql> use biginttest;
Database changed
mysql> describe oc_files_trash;
+-----------+--------------+------+-----+---------+----------------+
| Field     | Type         | Null | Key | Default | Extra          |
+-----------+--------------+------+-----+---------+----------------+
| auto_id   | int(11)      | NO   | PRI | NULL    | auto_increment |
| id        | varchar(250) | NO   | MUL |         |                |
| user      | varchar(64)  | NO   | MUL |         |                |
| timestamp | varchar(12)  | NO   | MUL |         |                |
| location  | varchar(512) | NO   |     |         |                |
| type      | varchar(4)   | YES  |     | NULL    |                |
| mime      | varchar(255) | YES  |     | NULL    |                |
+-----------+--------------+------+-----+---------+----------------+
7 rows in set (0.00 sec)

```

**Comment by PVince81** [comment]
@VicDeo better use "mysqldump" instead as it contains more details and also contains the indices.

**Comment by VicDeo** [comment]
```
mysql> use o1002;
Database changed
 show create table oc_federated_reshares;
| oc_federated_reshares | CREATE TABLE `oc_federated_reshares` (
  `share_id` int(11) NOT NULL,
  `remote_id` int(11) NOT NULL COMMENT 'share ID at the remote server',
  UNIQUE KEY `share_id_index` (`share_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |

mysql> use biginttest;
Database changed
mysql> show create table oc_federated_reshares;
| oc_federated_reshares | CREATE TABLE `oc_federated_reshares` (
  `share_id` bigint(20) NOT NULL,
  `remote_id` bigint(20) NOT NULL COMMENT 'share ID at the remote server',
  UNIQUE KEY `share_id_index` (`share_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |


mysql> use o1002;
Database changed
mysql> show create table oc_share_external;        
| oc_share_external | CREATE TABLE `oc_share_external` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `remote` varchar(512) COLLATE utf8_bin NOT NULL COMMENT 'Url of the remove owncloud instance',
  `remote_id` int(11) NOT NULL DEFAULT '-1',
  `share_token` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'Public share token',
  `password` varchar(64) COLLATE utf8_bin DEFAULT NULL COMMENT 'Optional password for the public share',
  `name` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'Original name on the remote server',
  `owner` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'User that owns the public share on the remote server',
  `user` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'Local user which added the external share',
  `mountpoint` varchar(4000) COLLATE utf8_bin NOT NULL COMMENT 'Full path where the share is mounted',
  `mountpoint_hash` varchar(32) COLLATE utf8_bin NOT NULL COMMENT 'md5 hash of the mountpoint',
  `accepted` int(11) NOT NULL DEFAULT '0',
  PRIMARY KEY (`id`),
  UNIQUE KEY `sh_external_mp` (`user`,`mountpoint_hash`),
  KEY `sh_external_user` (`user`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |

mysql> use biginttest;
Database changed
mysql> show create table oc_share_external;
oc_share_external | CREATE TABLE `oc_share_external` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `remote` varchar(512) COLLATE utf8_bin NOT NULL COMMENT 'Url of the remote owncloud instance',
  `remote_id` bigint(20) NOT NULL DEFAULT '-1',
  `share_token` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'Public share token',
  `password` varchar(64) COLLATE utf8_bin DEFAULT NULL COMMENT 'Optional password for the public share',
  `name` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'Original name on the remote server',
  `owner` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'User that owns the public share on the remote server',
  `user` varchar(64) COLLATE utf8_bin NOT NULL COMMENT 'Local user which added the external share',
  `mountpoint` varchar(4000) COLLATE utf8_bin NOT NULL COMMENT 'Full path where the share is mounted',
  `mountpoint_hash` varchar(32) COLLATE utf8_bin NOT NULL COMMENT 'md5 hash of the mountpoint',
  `accepted` int(11) NOT NULL DEFAULT '0',
  PRIMARY KEY (`id`),
  UNIQUE KEY `sh_external_mp` (`user`,`mountpoint_hash`),
  KEY `sh_external_user` (`user`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |

mysql> use o1002;
Database changed
mysql> show create table oc_files_trash;
| oc_files_trash | CREATE TABLE `oc_files_trash` (
  `auto_id` int(11) NOT NULL AUTO_INCREMENT,
  `id` varchar(250) COLLATE utf8_bin NOT NULL DEFAULT '',
  `user` varchar(64) COLLATE utf8_bin NOT NULL DEFAULT '',
  `timestamp` varchar(12) COLLATE utf8_bin NOT NULL DEFAULT '',
  `location` varchar(512) COLLATE utf8_bin NOT NULL DEFAULT '',
  `type` varchar(4) COLLATE utf8_bin DEFAULT NULL,
  `mime` varchar(255) COLLATE utf8_bin DEFAULT NULL,
  PRIMARY KEY (`auto_id`),
  KEY `id_index` (`id`),
  KEY `timestamp_index` (`timestamp`),
  KEY `user_index` (`user`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |

mysql> use biginttest;
Database changed
mysql> show create table oc_files_trash;
| oc_files_trash | CREATE TABLE `oc_files_trash` (
  `auto_id` int(11) NOT NULL AUTO_INCREMENT,
  `id` varchar(250) COLLATE utf8_bin NOT NULL DEFAULT '',
  `user` varchar(64) COLLATE utf8_bin NOT NULL DEFAULT '',
  `timestamp` varchar(12) COLLATE utf8_bin NOT NULL DEFAULT '',
  `location` varchar(512) COLLATE utf8_bin NOT NULL DEFAULT '',
  `type` varchar(4) COLLATE utf8_bin DEFAULT NULL,
  `mime` varchar(255) COLLATE utf8_bin DEFAULT NULL,
  PRIMARY KEY (`auto_id`),
  KEY `id_index` (`id`),
  KEY `timestamp_index` (`timestamp`),
  KEY `user_index` (`user`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |


```

**Comment by VicDeo** [comment]
Tests are passed. Merging.

**Comment by VicDeo** [comment]
Stable10: https://github.com/owncloud/core/pull/28581

**Pull request merged**